### ISSUE TITLE
Hard to find documentation for symbols function

### ISSUE THREAD

**Issue by SamUnimelb** [sympy/sympy#20505]
Hi there,

As a beginner learning sympy, I find it difficult to find clear documentation for the basic function "symbols". Yes, from https://docs.sympy.org/latest/gotchas.html#variables we can know something about the symbols function, but I bet it is not all.

By meaning "documentation", I mean a formal specification of the input variables, output variables and the functions of "symbols" can achieve, rather than just a number of example saying (and letting readers of your package) find out what all the usages of a function are. To give an example, I would sincerely suggest that we provide documentation such as: 

```
symbols(names,**kwargs) : A function that returns symbolic variables to be used in expressions, names means the python variables to be used as symbolic calculation variables. kwargs can be:

- sympy variable assumptions: such as real=True, for specifying assumptions of variables. see: https://docs.sympy.org/latest/modules/core.html#module-sympy.core.assumptions for details
- seq: Can be True or False, meaning whether to turn the named variables into a sequence of variable for iterations.
- etc.
```

Although I sincerely know symbols is a fundamental function for sympy, but there is no reason that formal official documentation is absent for this function.

**Comment by naveensaigit** [comment]
Please see the documentation [here](https://docs.sympy.org/latest/modules/core.html?highlight=symbols#sympy.core.symbol.symbols). It has a lot of examples

**Comment by sylee957** [comment]
I think that it needs a clear documentation as parameters-returns format than a single verbose tutorial.

**Comment by SamUnimelb** [comment]
> Please see the documentation [here](https://docs.sympy.org/latest/modules/core.html?highlight=symbols#sympy.core.symbol.symbols). It has a lot of examples

Yep, I have no issues with examples, what I mean are these documents need to specify inputs, targets and outputs for each function (and agree with @sylee957. I think sympy is pretty poor in this aspect. For these functions you think are clear, many of the usages are not clear to new users at all. To give you tips on how to write clear documentation, see:

https://www.naturaldocs.org/getting_started/documenting_your_code/
https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html

**Comment by oscarbenjamin** [comment]
Examples in the docs are good but there should be a clear statement of the definition and description of each parameter. Actually looking at those examples I didn't realise how many different ways the `symbols` function can be used. It looks to me like something over-complicated that would be hard to define.

**Comment by Mukulikaa** [comment]
Hello! Can I work on this issue considering all the inputs in this thread?
